arrays-and-slices.md

```diff
@@ -6,7 +6,7 @@ Arrays allow you to store multiple elements of the same type in a variable in
 a particular order.
 
 When you have an array, it is very common to have to iterate over them. So let's
-use [our new-found knowledge of `for`][for] to make a `Sum` function. `Sum` will
+use [our new-found knowledge of \`for\`](for.md) to make a `Sum` function. `Sum` will
```

Comment:
We shouldn't be escaping the backticks here, we wan thtem to be displayed like `for`

Reply:
Sorry, my fault. Before I changed it, it looked like this:

```
mdl .
arrays-and-slices.md:9: MD039 Spaces inside link text
```

I just fixed it in a bad way. Now I ignored this rule here. I think it's a problem of markdownlint, I'll feedback to the author later.

A small tip: You can install mdl to check the rules of your markdown docs. If you don't like to install it, you can push a temporary branch based this PR, and see what will happen in Travis CI. 😄